src/core/dbus-service.c

```diff
@@ -718,6 +747,55 @@ static int bus_service_set_transient_property(
         if (streq(name, "ReloadSignal"))
                 return bus_set_transient_reload_signal(u, name, &s->reload_signal, message, flags, error);
 
+        if (streq(name, "ExtraFileDescriptors")) {
+                int fd;
+                const char *fdname;
+
+                r = sd_bus_message_enter_container(message, 'a', "(hs)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(hs)", &fd, &fdname)) > 0) {
+                        _cleanup_(service_extra_fd_freep) ServiceExtraFD *efd = NULL;
+
+                        efd = new(ServiceExtraFD, 1);
+                        if (!efd)
+                                return -ENOMEM;
+
+                        *efd = (ServiceExtraFD) {
+                                .fd = -EBADF,
+                                .fdname = strdup(fdname),
+                        };
+
+                        if (!efd->fdname)
+                                return -ENOMEM;
+
+                        efd->fd = fcntl(fd, F_DUPFD_CLOEXEC, 3);
+                        if (efd->fd < 0)
+                                return -errno;
+
+                        /* Disallow empty string for ExtraFileDescriptor
+                        * Unlike OpenFile, StandardInput and friends, there isn't a good sane
+                        * default for an arbitrary FD
+                        */
+                        if (!fdname_is_valid(efd->fdname) || isempty(efd->fdname))
+                                return -EINVAL;
+
+                        if (UNIT_WRITE_FLAGS_NOOP(flags))
+                                continue;
```

Comment:
So I see the opposite in OpenFile where we want to allocate and validate the OpenFile first, meaning we allocate the object: https://github.com/systemd/systemd/blob/68f0c80bc4a9183b772e7cab68dc1aa85701534f/src/core/dbus-service.c#L693

Alternatively, I can just move 781-782 up top since validation is really only looking at the fdname

Reply:
That's what I did, resolving